src/components/NewMessage/NewMessage.vue

```diff
@@ -682,21 +689,7 @@ export default {
 			this.debouncedUpdateChatInput.clear()
 			this.chatExtrasStore.removeChatInput(this.token)
 
-			if (this.upload) {
-				// remove Quote component
-				this.chatExtrasStore.removeParentIdToReply(this.token)
-
-				if (this.$store.getters.getInitialisedUploads(this.currentUploadId).length) {
-					// If dialog contains files to upload, delegate sending
-					this.$emit('upload', { caption: this.text, options })
-					return
-				} else {
-					// Dismiss dialog, process as normal message sending otherwise
-					this.$emit('sent')
-				}
-			}
-
-			if (this.hasText) {
+			if (this.hasText || (this.dialog && this.upload)) {
```

Comment:
`dialog` is sufficient because each `upload` is `dialog` in our approach (and this assumption is already used in removing non-dialog components of NewMessage like out of office)

Reply:
I'd keep it as-is, since we have dialogs - non-uploads. That change would allow broadcast empty strings or post empty messages to BR, which we don't want to have